stacker/builder.py

```diff
@@ -142,15 +146,20 @@ def conn(self):
         return self._conn
 
     @property
+    def s3_conn(self):
+        if not getattr(self, '_s3_conn', None):
```

Comment:
switched this to "hasattr", the point is that this is a private variable and should only be accessed by the property to cache the connection.